util/src/rate_counter.rs

```diff
@@ -105,6 +105,6 @@ impl RateCounter {
 fn millis_since_epoch() -> u64 {
 	let since_epoch = SystemTime::now()
 		.duration_since(SystemTime::UNIX_EPOCH)
-		.unwrap_or(Duration::new(0, 0));
+		.unwrap_or_else(|_| Duration::new(0, 0));
```

Comment:
Trivial comment but why do this? Is it to avoid calling Duration::new(0,0) each time regardless?
If so maybe better to turn this into a constant and reuse it in a cleaner `unwrap_or()`?

Reply:
Good point! 
From rust doc:

>Arguments passed to `unwrap_or` are eagerly evaluated; if you are passing the result of a function call, it is recommended to use `unwrap_or_else`, which is lazily evaluated

Actually we can simplify this since Rust 1.33. Will update. (https://stackoverflow.com/questions/26593387/how-can-i-get-the-current-time-in-milliseconds)